#!/usr/bin/env python
PACKAGE = "object_detection"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("minBlobSize",     int_t,    0, "min blob size", 50,  0,  5000)
gen.add("maxBlobSize",     int_t,    0, "max blob size", 1000000,  0,  1000000)
gen.add("minCircularity",     double_t,    0, "Minimal Circularity", 0.02,  0, 1.0)
gen.add("minRoundness",     double_t,    0, "Minimal Roundness", 0.95,  0, 5.0)
gen.add("objectDiameter",     double_t,    0, "Object diameter [m]", 0.2,  0, 0.5)
gen.add("histogramScale",     int_t,    0, "Histogram image scaling", 2,  1, 5)
gen.add("visualDistanceToleranceRatio",     double_t,    0, "Visual tolerance (ratio)", 0.3,  0, 2.0)
gen.add("visualDistanceToleranceAbsolute",     double_t,    0, "Visual tolerance (absolute)", 0.5,  0, 2.0)
gen.add("longObject",     bool_t,    0, "Detect long objects", False,  False, True)

exit(gen.generate(PACKAGE, "object_detection", "object_detection"))
